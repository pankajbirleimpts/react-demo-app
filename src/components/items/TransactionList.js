import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Grid } from 'semantic-ui-react';
import { getAllTransactions } from '../../actions/ItemAction';
import { Loader, CustomTable } from '../common';

class TransactionList extends Component {
  constructor(props) {
    super(props);
    this.state = {
      columns: [
        {
          label: 'User Name',
          keyName: 'userName',
          sort: true,
          searchable: true,
          sortable: true
        },
        {
          label: 'Item name',
          keyName: 'itemName',
          sort: null,
          searchable: true,
          sortable: true
        },
        {
          label: 'Category',
          keyName: 'category',
          sort: null,
          searchable: true,
          sortable: true
        },
        {
          label: 'Date',
          keyName: 'date',
          sort: null,
          searchable: true,
          sortable: true
        },
        {
          label: 'Rate',
          keyName: 'amount',
          sort: null,
          searchable: true,
          sortable: true
        },
        {
          label: 'Quantity',
          keyName: 'quantity',
          sort: null,
          searchable: true,
          sortable: true
        },
        {
          label: 'Total Amount',
          keyName: 'purchaseAmount',
          sort: null,
          searchable: true,
          sortable: true
        }
      ]
    };
  }

  componentDidMount() {
    const { user } = this.props;
    const userId = user.data.role === 'EMPLOYEE' ? user.data.id : '';
    this.props.getAllTransactions(userId);
  }


  render() {
    return (
      <Grid>
        <Loader isLoading={this.props.item.isLoading} />
        <Grid.Row centered>
          <CustomTable
            columns={this.state.columns}
            tableData={this.props.item.allTransactions}
            module='TRANSACTIONS'
          />
        </Grid.Row>
      </Grid>
    );
  }
}

function mapStateToProp({ item, user }) {
  return {
    item,
    user
  };
}

export default connect(mapStateToProp, { getAllTransactions })(
  TransactionList
);
