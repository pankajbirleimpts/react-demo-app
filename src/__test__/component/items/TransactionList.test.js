import React from "react";
import { shallow } from "enzyme";
import TransactionList from "../../../components/items/TransactionList";
import { storeFactory } from "../../test/TestUtils";

const setup = (props, actions = null) => {
  const customActions = {
    getAllTransactions: jest.fn(),
    ...actions
  };
  const store = storeFactory(props);
  const wrapper = shallow(<TransactionList store={store} {...customActions} />, { lifecycleExperimental: true })
    .dive()
    .dive();
  return wrapper;
};

describe("TransactionList component", () => {
  const props = {
    user: {
      data: {
        balance: 4955,
        country: "Indore",
        email: "pankaj.birle@impetus.co.in",
        employeeId: "IIIPL3025",
        firstName: "Pankaj",
        id: "-Lx1WclrnUuv0_hSvdpV",
        lastName: "Birle",
        password: "pankaj123",
        role: "EMPLOYEE"
      }
    },
    item: {
      isLoading: false,
      allTransactions: [
        {
          amount: 15,
          date: "26-Dec-2019",
          itemId: "-Lwi9T6G70Fs-Wd5JF0r",
          purchaseAmount: "0.00",
          quantity: 0,
          userId: "-Lwc-iXIVHhQnEihsqcW",
          id: "-Lx18EC9oMqwARQ2HL5m",
          itemName: "Tawa Roti",
          category: "Roti",
          userName: "Harish Nararayan"
        },
        {
          amount: 15,
          date: "26-Dec-2019",
          itemId: "-Lwi9T6G70Fs-Wd5JF0r",
          purchaseAmount: "10.00",
          quantity: 1,
          userId: "-Lwc-iXIVHhQnEihsqcs",
          id: "-Lx18EC9oMqwARQ2HL5n",
          itemName: "Tawa Roti",
          category: "Roti",
          userName: "Harish Nararayan"
        }
      ]
    }
  };
  test("Snapshot of TransactionList component", () => {
    const wrapper = setup(props);
    expect(wrapper).toMatchSnapshot();
  });

  test("Check the isLoading in props", () => {
    const wrapper = setup(props);
    const componentProps = wrapper.instance().props;
    expect(componentProps.item.isLoading).toBe(props.item.isLoading);
  });

  test("Check the 'EMPLOYEE' role in props", () => {
    const wrapper = setup(props);
    const componentProps = wrapper.instance().props;
    expect(componentProps.user.data.role).toBe(props.user.data.role);
  });

  test("Check the `allTransactions` in props", () => {
    const wrapper = setup(props);
    const componentProps = wrapper.instance().props;
    expect(componentProps.item.allTransactions).toBe(
      props.item.allTransactions
    );
  });

  test("Check the `getAllTransactions` action in props", () => {
    const wrapper = setup(props);
    const componentProps = wrapper.instance().props;
    expect(componentProps.getAllTransactions).toBeInstanceOf(Function);
  });
  
});
