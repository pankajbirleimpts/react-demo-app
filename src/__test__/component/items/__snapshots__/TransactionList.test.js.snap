// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionList component Snapshot of TransactionList component 1`] = `
<Grid>
  <Loading
    isLoading={false}
  />
  <GridRow
    centered={true}
  >
    <Connect(UnconnectedCustomTable)
      columns={
        Array [
          Object {
            "keyName": "userName",
            "label": "User Name",
            "searchable": true,
            "sort": true,
            "sortable": true,
          },
          Object {
            "keyName": "itemName",
            "label": "Item name",
            "searchable": true,
            "sort": null,
            "sortable": true,
          },
          Object {
            "keyName": "category",
            "label": "Category",
            "searchable": true,
            "sort": null,
            "sortable": true,
          },
          Object {
            "keyName": "date",
            "label": "Date",
            "searchable": true,
            "sort": null,
            "sortable": true,
          },
          Object {
            "keyName": "amount",
            "label": "Rate",
            "searchable": true,
            "sort": null,
            "sortable": true,
          },
          Object {
            "keyName": "quantity",
            "label": "Quantity",
            "searchable": true,
            "sort": null,
            "sortable": true,
          },
          Object {
            "keyName": "purchaseAmount",
            "label": "Total Amount",
            "searchable": true,
            "sort": null,
            "sortable": true,
          },
        ]
      }
      module="TRANSACTIONS"
      tableData={
        Array [
          Object {
            "amount": 15,
            "category": "Roti",
            "date": "26-Dec-2019",
            "id": "-Lx18EC9oMqwARQ2HL5m",
            "itemId": "-Lwi9T6G70Fs-Wd5JF0r",
            "itemName": "Tawa Roti",
            "purchaseAmount": "0.00",
            "quantity": 0,
            "userId": "-Lwc-iXIVHhQnEihsqcW",
            "userName": "Harish Nararayan",
          },
          Object {
            "amount": 15,
            "category": "Roti",
            "date": "26-Dec-2019",
            "id": "-Lx18EC9oMqwARQ2HL5n",
            "itemId": "-Lwi9T6G70Fs-Wd5JF0r",
            "itemName": "Tawa Roti",
            "purchaseAmount": "10.00",
            "quantity": 1,
            "userId": "-Lwc-iXIVHhQnEihsqcs",
            "userName": "Harish Nararayan",
          },
        ]
      }
    />
  </GridRow>
</Grid>
`;
